Don't allow non-numeric priority tags

jmsanders opened the pull request:
We need these to be sortable and if we prevent this at tag creation time, we could avoid introducing this potentially expensive regex calculation:

https://github.com/dagster-io/dagster/pull/18172

alangenfeld commented on python_modules/dagster/dagster/_core/definitions/utils.py:
```diff
@@ -96,6 +96,14 @@ def validate_tags(
     for key, value in check.opt_mapping_param(tags, "tags", key_type=str).items():
+        if key == PRIORITY_TAG:
+            try:
+                int(value)
```
how explicit are we that these should be `int` instead of `float` ? It looks like strings-of-floats don't coerce to `int` even though actual floats do:

```
>>> int(100.5)
100
>>> int("100.5")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '100.5'
```

does the current set-up silently treat floats as bad priorities?

alangenfeld reviewed (approved):
would be good to ensure update the end to end test to make sure floats work in the new DB scheme as well

jmsanders commented:
I added a test to make sure the tag validation itself works. I don't expect this to have any impact on the db schema - all of our tags are just stored as some variation of `db.Text`.

But I've also updated https://github.com/dagster-io/dagster/pull/18125 to sort on floats instead of ints which I think gives us sufficient coverage here.

jmsanders commented:
Closed eventually in favor of #18347
